x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -433,6 +433,7 @@ public boolean test(PersistentTasksCustomMetaData.PersistentTask<?> persistentTa
                 case STOPPING:
                     exception = ExceptionsHelper.conflictStatusException("the task has been stopped while waiting to be started");
                     return true;
+                case STARTING:
```

Comment:
I think this means that when a user calls start with `lazy` mode they would not get a response until the task is actually assigned, right? We probably don't want that.

Reply:
In the case where the job is too big to be assigned this check higher up the method will return early:

```
            // This means we are awaiting a new node to be spun up, ok to return back to the user to await node creation
            if (assignment != null && assignment.equals(JobNodeSelector.AWAITING_LAZY_ASSIGNMENT)) {
                return true;
            }
```

The only time the code will get to the `STARTING` case in the `switch` is during the tiny fraction of a second when a job has been successfully assigned but the call to update its task state has not yet completed.  Previously in this situation the state was reported as `STOPPED`, so I thought it best to put `STARTING` alongside `STOPPED` in the `switch` to maintain previous behaviour.

(I think the integration tests confirm this works as I described, but just to double check I tested this manually with a job with a `3tb` `model_memory_limit` and `allow_lazy_start: true` on my laptop and the start endpoint returned success instantly.)

It's equivalent to this bit of `TransportOpenJobAction`: https://github.com/elastic/elasticsearch/blob/f52f2aef9707bdf93d9a785452429137d2689b5a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportOpenJobAction.java#L561-L563

But since it's clearly confusing I'll add some comments.

Reply:
Right, got it.